compute/disk-images/create_disk_image.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snippets
+
+// [START compute_windows_image_create]
+// [START compute_images_create]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+// Creates a disk image from an existing disk
+func createImageFromDisk(
+	w io.Writer,
+	projectID, zone, sourceDiskName, imageName string,
+	storageLocations []string,
+	forceCreate bool,
+) error {
+	// projectID := "your_project_id"
+	// zone := "us-central1-a"
+	// sourceDiskName := "your_disk_name"
+	// imageName := "my_image"
+	// // If storageLocations empty, automatically selects the closest one to the source
+	// storageLocations = []string{}
+	// // If forceCreate is set to `true`, proceeds even if the disk is attached to
+	// // a running instance. This may compromise integrity of the image!
+	// forceCreate = false
+
+	ctx := context.Background()
+	disksClient, err := compute.NewDisksRESTClient(ctx)
```

Comment:
Do you need the REST client? Or is there a gRPC client you can use?

Reply:
There is no gRPC client for Compute APIs. Compute Engine APIs predate gRPC rollout in the Cloud.

This causes all sorts of issues and additional complexity because most of the tooling expects gRPC (documentation generation, etc) so we generate fake protos from REST APIs to interoperate with the tooling. But that's fortunately irrelevant here.